src/SimpleThings/AppBundle/Entity/Result.php

```
@@ -0,0 +1,185 @@
+     * @param string $code
+     * @return null|Metric
+     */
+    public function getMetric($code)
+    {
+        $metrics = $this->getMetrics();
+
+        return isset($metrics[$code]) ? $metrics[$code] : null;
+    }
+
+    /**
+     * @param string $code
+     * @return bool
+     */
+    public function hasMetric($code)
+    {
+        $metrics = $this->getMetrics();
+
+        return isset($metrics[$code]);
+    }
+
+    /**
+     * @param Issue $issue
+     * @throws \Exception
+     */
+    public function addIssue(Issue $issue)
+    {
+        throw new \Exception('Readonly!');
+    }
+
+    /**
+     * @param Metric $metric
+     * @throws \Exception
+     */
+    public function addMetric(Metric $metric)
+    {
+        throw new \Exception('Readonly!');
+    }
+
+    /**
+     * @param BaseResult $result
+     * @throws \Exception
+     */
+    public function merge(BaseResult $result)
+    {
+        throw new \Exception('Readonly!');
+    }
+
+    /**
+     * @param Metric[] $metrics
+     * @return Metric[]
+     */
+    private function prepareMetrics(array $metrics)
+    {
+        $return = [];
+
+        foreach ($metrics as $metric) {
+            $return[$metric->getCode()] = $metric;
+        }
+
+        return $return;
+    }
+
+    /**
+     * @param array $data
+     * @return string
+     */
+    private function serialize($data)
+    {
+        $json = $this->getSerializer()->serialize($data, 'json');
+
+        return $this->getCompressor()->compress($json);
+    }
+
+    /**
+     * @param resource $data
+     * @param string $class
+     * @return array
+     */
+    private function deserialize($data, $class)
+    {
+        if (!is_resource($data)) {
+            return [];
+        }
+
+        $compressedData = stream_get_contents($data);
+
+        if (!$compressedData) {
+            return [];
+        }
+
+        $json = $this->getCompressor()->uncompress($compressedData);
+
+        return (array)$this->getSerializer()
```

does this fail silently or throw exceptions?